pkg/agent/apiserver/handlers/getpodinterface/handler.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package getpodinterface
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/interfacestore"
+	"github.com/vmware-tanzu/antrea/pkg/monitor"
+)
+
+// GetPodInterfaceResponse describes the internal response struct of the pod command.
+type GetPodInterfaceResponse struct {
+	PodName         string                            `json:"name" yaml:"name" antctl:"key,The name of the pod"`
+	InterfaceConfig []*interfacestore.InterfaceConfig `json:",inline" yaml:",inline"`
+}
+
+// HandleFunc returns the function which can handle queries issued by the pod command,
+func HandleFunc(aq monitor.AgentQuerier) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if aq == nil {
+			w.WriteHeader(http.StatusNotImplemented)
+			return
+		}
+
+		ns, _ := r.URL.Query()["namespace"]
+
+		key, doFilter := r.URL.Query()["name"]
+		if doFilter && (len(key) != 1 || len(key[0]) == 0) {
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		m := make(map[string][]*interfacestore.InterfaceConfig)
+		for _, v := range aq.GetInterfacesByType(interfacestore.ContainerInterface) {
+			podName := (*v.ContainerInterfaceConfig).PodName
+			vPodNS := (*v.ContainerInterfaceConfig).PodNamespace
+			if (!doFilter || key[0] == podName) && ns[0] == vPodNS {
+				m[podName] = append(m[podName], v)
+			}
+			fmt.Println("podName:", podName, "vPodNS:", vPodNS)
+		}
+
+		if doFilter {
+			if len(m) == 0 {
+				w.WriteHeader(http.StatusNotFound)
+			} else {
+				var pod GetPodInterfaceResponse
+				for k, v := range m {
+					pod = GetPodInterfaceResponse{k, v}
+					break
+				}
+				err := json.NewEncoder(w).Encode(pod)
+				if err != nil {
+					w.WriteHeader(http.StatusInternalServerError)
+				}
+			}
+		} else {
+			var pods []GetPodInterfaceResponse
+			for k, v := range m {
+				pods = append(pods, GetPodInterfaceResponse{k, v})
+			}
+			err := json.NewEncoder(w).Encode(pods)
+			if err != nil {
+				w.WriteHeader(http.StatusInternalServerError)
+			}
+		}
```

Comment:
This can be simpler:
```
if doFilter and len(m) == 0 {
	w.WriteHeader(http.StatusNotFound)
} else {
	var pods []GetPodInterfaceResponse
	for k, v := range m {
		pods = append(pods, GetPodInterfaceResponse{k, v})
	}
	err := json.NewEncoder(w).Encode(pods)
	if err != nil {
		w.WriteHeader(http.StatusInternalServerError)
	}
}
```

Reply:
Refactored.